spring-boot-project/spring-boot/src/main/java/org/springframework/boot/DigitalAmount.java

```diff
@@ -0,0 +1,307 @@
+	 */
+	public boolean isNegative() {
+		return this.amount < 0;
+	}
+
+	/**
+	 * Checks if this duration is {@code DigitalAmount} positive, excluding zero.
+	 * @return true if this {@code DigitalAmount} has a total amount greater than zero
+	 */
+	public boolean isPositive() {
+		return this.amount > 0;
+	}
+
+	/**
+	 * Checks if this {@code DigitalAmount} is zero.
+	 * @return true if this {@code DigitalAmount} has a total amount equal to zero
+	 */
+	public boolean isZero() {
+		return this.amount == 0;
+	}
+
+	/**
+	 * If the value matches the given predicate, return an {@code Optional} describing the
+	 * value, otherwise return an empty {@code Optional}.
+	 * @param predicate a predicate to apply to the value.
+	 * @return an {@code Optional} describing the value if a value matches the given
+	 * predicate, otherwise an empty {@code Optional}
+	 */
+	public Optional<DigitalAmount> filter(
+			@NonNull Predicate<? super DigitalAmount> predicate) {
+		Assert.notNull(predicate, () -> "Predicate must not be null");
+		if (predicate.test(this)) {
+			return Optional.of(this);
+		}
+		return Optional.empty();
+	}
+
+	/**
+	 * Apply the provided mapping function to this.
+	 * @param <T> the type of the result of the mapping function
+	 * @param mapper a mapping function to apply to the value.
+	 * @return the result of applying a mapping function
+	 */
+	public <T> T map(@NonNull Function<? super DigitalAmount, ? extends T> mapper) {
+		Assert.notNull(mapper, () -> "Mapper must not be null");
+		return mapper.apply(this);
+	}
+
+	/**
+	 * Returns a {@code DigitalAmount} whose value is {@code (this +
+	 * other)}.
+	 * @param amount value to be added to this {@code DigitalAmount}.
+	 * @param unit {@code DigitalUnit} of this amount.
+	 * @return {@code this + other}
+	 */
+	public DigitalAmount add(long amount, @NonNull DigitalUnit unit) {
+		return add(fromUnit(amount, unit));
+	}
+
+	/**
+	 * Returns a {@code DigitalAmount} whose value is {@code (this -
+	 * other)}.
+	 * @param amount value to be subtracted to this {@code DigitalAmount}.
+	 * @param unit {@code DigitalUnit} of this amount.
+	 * @return {@code this - other}
+	 */
+	public DigitalAmount subtract(long amount, @NonNull DigitalUnit unit) {
+		return subtract(fromUnit(amount, unit));
+	}
+
+	/**
+	 * Returns a new {@code DigitalAmount} whose value is {@code (this +
+	 * other)}.
+	 * @param other value to be added to this {@code DigitalAmount}.
+	 * @return {@code this + other}
+	 */
+	public DigitalAmount add(@NonNull DigitalAmount other) {
+		Assert.notNull(other, () -> "Digital Amount must not be null");
+		return new DigitalAmount(Math.addExact(this.amount, other.amount));
+	}
+
+	/**
+	 * Returns a new {@code DigitalAmount} whose value is {@code (this -
+	 * other)}.
+	 * @param other value to be subtracted from this {@code DigitalAmount}.
+	 * @return {@code this - other}
+	 */
+	public DigitalAmount subtract(@NonNull DigitalAmount other) {
```

Comment:
All the methods above should go away please. There is really no need to start with something that complex/complete. The purpose of the enhancement is to have a data type to represent a "data size" for binding purposes.